src/compiler/checker.ts

```diff
@@ -12836,6 +12836,42 @@ namespace ts {
                         }
                     }
                 }
+                else if (target.flags & TypeFlags.Conditional) {
+                    const c = target as ConditionalType;
+                    let skipFalse = false;
+                    let skipTrue = false;
+                    // Check if the conditional is always true or always false but still deferred for distribution purposes
+                    if (!isTypeAssignableTo(getPermissiveInstantiation(c.checkType), getPermissiveInstantiation(c.extendsType))) {
+                        skipTrue = true;
+                    }
+                    else if (isTypeAssignableTo(getRestrictiveInstantiation(c.checkType), getRestrictiveInstantiation(c.extendsType))) {
+                        skipFalse = true;
+                    }
+
+                    // Instantiate with a replacement mapper if the conditional is distributive, replacing the check type with a clone of itself,
+                    // this way {x: string | number, y: string | number} -> (T extends T ? { x: T, y: T } : never) appropriately _fails_ when
+                    // T = string | number (since that will end up distributing and producing `{x: string, y: string} | {x: number, y: number}`,
+                    // to which `{x: string | number, y: string | number}` isn't assignable)
+                    let distributionMapper: TypeMapper | undefined;
+                    const checkVar = getActualTypeVariable(c.root.checkType);
+                    if (c.root.isDistributive && checkVar.flags & TypeFlags.TypeParameter) {
+                        const newParam = cloneTypeParameter(checkVar);
+                        distributionMapper = createReplacementMapper(checkVar, newParam, c.mapper || identityMapper);
+                        newParam.mapper = distributionMapper;
+                    }
+
+                    // TODO: Find a nice way to include potential conditional type breakdowns in error output, if those seems good (they usually don't)
+                    let localResult: Ternary | undefined;
+                    if (skipTrue || (localResult = isRelatedTo(source, instantiateType(c.trueType, distributionMapper), /*reportErrors*/ false))) {
+                        if (!skipFalse) {
+                            localResult = (localResult || -1) & isRelatedTo(source, instantiateType(c.falseType, distributionMapper), /*reportErrors*/ false);

```

Comment:
`Ternary.True`?

Reply:
Naw, but explcitly writing `Ternary.Maybe` would be better. Defaulting to `Maybe` when the first check is skipped is what we want here, this way if the second check is `Maybe` we propegate the `Maybe`.